### DISCUSSION TITLE
[Feature] Parsing of AI Generated Images Metadata -> Tags

### DISCUSSION THREAD

**Discussion by xedis** [immich-app/immich#12525] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi! 
First off, love the app it's awesome. I've begun importing the nearing 1M images I've generated with SD.Next/Stable-diffusion/etc 

Second, it would be insanely helpful if the metadata stored in these images could be leveraged for tags. It's actually quite robust how much information these frontends for stable diffuision include written in the png exifdata, if you know where to look  and how each UI formats this data. Sadly, there has yet to emerge a Standard across the popular platforms but they all store it in roughly the same area, though the length of it can be excessive (such as with ComfyUI) 

I'm including a link to the parsing code snippets for one of the best image managers that directly integrates with all of the popular front-ends for image gen. Maybe it will make integration of this information easier. 

[Stable-diffusion-webui/Inifinite Image Browser-extension Parsing
](https://github.com/zanllp/sd-webui-infinite-image-browsing/tree/24443baf3e3091457067c7484b3539c845d68159/scripts/iib/parsers)

I can provide further context if you guys think it's something that could be implemented. Able to provide image samples for testing of proper parsing and I'm an intermitently active end-user support person for the SDNext discord. 

Thanks for all your hard work and the love you guys have put into this application, it really shows. 

Xedis

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Immich isn't aimed at AI generated content, so we don't plan to add support for this (especially since it's non-standard). Eventually we aim to have plugin support, and then this can be implemented by the community that way.

**Comment by DeepaMalik12** [comment]
This sounds like an incredibly useful feature! Utilizing the metadata from AI-generated images to automatically generate tags would significantly reduce manual effort, especially for large libraries. Although there isn't a universal standard yet, it seems completely feasible to leverage the common EXIF fields from various SD frontends. It would be fantastic if the app could incorporate your parsing code to simplify the organization and search of images

**Reply by ThesiiNCey** [reply]
true this would be usefull
